artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/SharedEventLoopGroup.java

```diff
@@ -84,30 +84,32 @@ public ThreadFactory run() {
 
    @Override
    public Future<?> shutdownGracefully(final long l, final long l2, final TimeUnit timeUnit) {
-      if (channelFactoryCount.decrementAndGet() == 0) {
-         shutdown.compareAndSet(null, next().scheduleAtFixedRate(new Runnable() {
-            @Override
-            public void run() {
-               synchronized (SharedEventLoopGroup.class) {
-                  if (shutdown.get() != null) {
-                     Future<?> future = SharedEventLoopGroup.super.shutdownGracefully(l, l2, timeUnit);
-                     future.addListener(new FutureListener<Object>() {
-                        @Override
-                        public void operationComplete(Future<Object> future) throws Exception {
-                           if (future.isSuccess()) {
-                              terminationPromise.setSuccess(null);
-                           } else {
-                              terminationPromise.setFailure(future.cause());
+      synchronized (SharedEventLoopGroup.class) {
```

Comment:
if you applied the synchronize here to this instead of .class I would be ok with merging this without a test. 

A synchornize on a .class here.. without a test.. I don't know what implications it would have.

Can you tell me why the synchornize on the class was needed?

Reply:
Thats a good point there is risk especially where you may run two seperate clients to different brokers or have multiple embedded brokers. Sync on class or static sharing isnt always safe